src/Symfony/Component/Messenger/Tests/DependencyInjection/MessengerPassTest.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Tests\DependencyInjection;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Argument\ServiceClosureArgument;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+use Symfony\Component\Messenger\ContainerHandlerLocator;
+use Symfony\Component\Messenger\DependencyInjection\MessengerPass;
+use Symfony\Component\Messenger\Tests\Fixtures\DummyMessage;
+use Symfony\Component\Messenger\Transport\ReceiverInterface;
+
+class MessengerPassTest extends TestCase
+{
+    public function testProcess()
+    {
+        $container = $this->getContainerBuilder();
+        $container
+            ->register(DummyHandler::class, DummyHandler::class)
+            ->addTag('messenger.message_handler')
+        ;
+        $container
+            ->register(DummyReceiver::class, DummyReceiver::class)
+            ->addTag('messenger.receiver')
+        ;
+
+        (new MessengerPass())->process($container);
+
+        $handlerLocatorDefinition = $container->getDefinition($container->getDefinition('messenger.handler_resolver')->getArgument(0));
+        $this->assertSame(ServiceLocator::class, $handlerLocatorDefinition->getClass());
+        $this->assertEquals(
+            array('handler.'.DummyMessage::class => new ServiceClosureArgument(new Reference(DummyHandler::class))),
+            $handlerLocatorDefinition->getArgument(0)
+        );
+
+        $this->assertEquals(
+            array(DummyReceiver::class => new Reference(DummyReceiver::class)),
+            $container->getDefinition('messenger.receiver_locator')->getArgument(0)
+        );
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
```

Comment:
Not sure what's our stance on this, but [phpunit advices to use method calls](https://thephp.cc/news/2016/02/questioning-phpunit-best-practices) instead of annotations.

Reply:
for now we use annotations everywhere, perhaps it should be rediscussed on 2.7